qldbdriver/result.go

```diff
@@ -66,9 +68,51 @@ func (result *Result) getNextPage() error {
 	result.pageValues = nextPage.Page.Values
 	result.pageToken = nextPage.Page.NextPageToken
 	result.index = 0
+	result.updateMetrics(nextPage)
 	return nil
 }
 
+func (result *Result) updateMetrics(fetchPageResult *qldbsession.FetchPageResult) {
+	if fetchPageResult.ConsumedIOs != nil {
+		if result.consumedIOs == nil {
+			result.consumedIOs = &IOUsage{new(int64), new(int64)}
+		}
+		*result.consumedIOs.readIOs += *fetchPageResult.ConsumedIOs.ReadIOs
+		*result.consumedIOs.writeIOs += *fetchPageResult.ConsumedIOs.WriteIOs
+	}
+
+	if fetchPageResult.TimingInformation != nil {
+		if result.timingInformation == nil {
+			result.timingInformation = &TimingInformation{new(int64)}
+		}
+		*result.timingInformation.processingTimeMilliseconds += *fetchPageResult.TimingInformation.ProcessingTimeMilliseconds
+	}
+}
+
+// GetConsumedIOs returns the statement statistics for the current number of read IO requests. The statistics are stateful.
+func (result *Result) GetConsumedIOs() *IOUsage {
+	if result.consumedIOs == nil {
+		return nil
+	}
+	readIOs := *result.consumedIOs.readIOs
+	writeIOs := *result.consumedIOs.writeIOs
```

Comment:
We create these variables instead of passing the `result.consumed.readIO/writeIO` pointers to the IOUsage struct because the result is stateful. 

We don't want to give a struct containing the result pointer which will update as it iterates through the values.